hail/python/hail/typecheck/check.py

```diff
@@ -138,7 +138,7 @@ def check(self, x, caller, param):
         return x_
 
     def expects(self):
-        return 'dict[%s, %s]' % (self.kc.expects(), self.vc.expects())
+        return 'MutableMapping[%s, %s]' % (self.kc.expects(), self.vc.expects())
```

Comment:
@tpoterba can you verify you're ok with `dictof` checking for a `MutableMapping` rather than a literal dict?

Reply:
I'd be fine with just `Mapping`.

Generally mutability is not something we really want/care about

Reply:
Even better on both counts.  Fixed, thanks!